index.js

```diff
@@ -317,3 +317,450 @@ const arrayLeaders = arr => {
+  let total = 0;
+  let calls = 0;
+  return function(n) {
+    calls++;
+    total += n;
+    return Math.round((total / calls) * 100) / 100;
+  };
+}
+
+//Task 47
+
+const stringExpansion = str => {
+  return str.replace(/(\d)([^\d]*)/g, function(match) {
+    return match.slice(1).repeat(match[0]);
+  });
+};
+
+//Task 48
+
+const followTheSpy = arr => {
+  let correctRoute = [];
+  arr
+    .map(elem => {
+      if (!arr.find(country => country[1] === elem[0])) {
+        correctRoute.push(elem[0]);
+        correctRoute.push(elem[1]);
+      }
+    })
+    .map(() => {
+      arr.map(elem => {
+        elem[0] === correctRoute[correctRoute.length - 1] &&
+          correctRoute.push(elem[1]);
+      });
+    });
+  return correctRoute.join(",");
+};
+
+//Task 49
+
+const humanYearsCatYearsDogYears = humYears => {
+  let yearsArr = [humYears, 0, 0];
+  if (humYears === 1) {
+    yearsArr[1] = 15;
+    yearsArr[2] = 15;
+  } else if (humYears === 2) {
+    yearsArr[1] = 24;
+    yearsArr[2] = 24;
+  } else if (humYears >= 3) {
+    yearsArr[1] = 24 + 4 * (humYears - 2);
+    yearsArr[2] = 24 + 5 * (humYears - 2);
+  }
+  return yearsArr;
+};
+
+//Task 50
+
+String.prototype.camelCase = function() {
+  return this.split(" ")
+    .map(elem => {
+      return elem.charAt(0).toUpperCase() + elem.slice(1);
+    })
+    .join("");
+};
+
+//Task 51
+
+const abbreviatedName = name => {
+  return name
+    .split(" ")
+    .map((elem, i) => {
+      if (i > 0) {
+        return elem.charAt(0) + ".";
+      }
+      return elem;
+    })
+    .join(" ");
+};
+
+//Task 52
+
+const sumAll = arr => {
+  if (arr[1] > arr[0]) {
+    let summArr = [];
+    let counter = arr[0] - 1;
+    for (let i = arr[0]; i <= arr[1]; i++) {
+      counter++;
+      summArr.push(counter);
+    }
+    return summArr.reduce((a, b) => a + b);
+  } else if (arr[0] > arr[1]) {
+    let summArr = [];
+    let counter = arr[1] - 1;
+    for (let i = arr[1]; i <= arr[0]; i++) {
+      counter++;
+      summArr.push(counter);
+    }
+    return summArr.reduce((a, b) => a + b);
+  }
+};
+
+//Task 53
+
+const diffArray = (arr1, arr2) => {
+  let filterArr1 = arr1.filter(x => arr2.indexOf(x) === -1);
+  let filterArr2 = arr2.filter(x => arr1.indexOf(x) === -1);
+  return filterArr1.concat(filterArr2);
+};
+
+//Task 54
+
+const convertToRoman = num => {
+  let lookup = {
+    M: 1000,
+    CM: 900,
+    D: 500,
+    CD: 400,
+    C: 100,
+    XC: 90,
+    L: 50,
+    XL: 40,
+    X: 10,
+    IX: 9,
+    V: 5,
+    IV: 4,
+    I: 1
+  };
+  let roman = "";
+  for (let i in lookup) {
+    while (num >= lookup[i]) {
+      roman += i;
+      num -= lookup[i];
+    }
+  }
+  return roman;
+};
+
+//Task 55
+
+const whatIsInAName = (collection, source) => {
+  let arr = [];
+  let keys = Object.keys(source);
+  let = collection.filter(obj => {
+    return keys.every(key => {
+      return obj.hasOwnProperty(key) && obj[key] === source[key];
+    });
+  });
+  return arr;
+};
+
```

Comment:
Please remove this lines, because these tasks are not relevant to this branch

Reply:
// Task 56 

+ a: LVI ✊ 7:
+ b: II
+ c: @@✊✌️✊
+ d: (L}…{OW}{N)ii

A: (LIV) ✊ 

BiD 1♥️